driver-core/src/main/com/mongodb/operation/MixedBulkWriteOperation.java

```diff
@@ -440,15 +441,34 @@ private void executeCommandAsync(final AsyncWriteBinding binding, final AsyncCon
                                      final SingleResultCallback<BsonDocument> commandCallback) {
         try {
             connection.commandAsync(namespace.getDatabaseName(), batch.getCommand(), NO_OP_FIELD_NAME_VALIDATOR,
-                    null, batch.getDecoder(), binding.getSessionContext(), shouldAcknowledge(batch, writeConcern),
+                    null, batch.getDecoder(), binding.getSessionContext(), shouldAcknowledge(batch, binding.getSessionContext()),
                     batch.getPayload(), batch.getFieldNameValidator(), commandCallback);
         } catch (Throwable t) {
             callback.onResult(null, t);
         }
     }
 
-    private boolean shouldAcknowledge(final BulkWriteBatch batch, final WriteConcern writeConcern) {
-        return ordered ? batch.hasAnotherBatch() || writeConcern.isAcknowledged() : writeConcern.isAcknowledged();
+
+    private WriteConcern getAppliedWriteConcern(final WriteBinding binding) {
+        return getAppliedWriteConcern(binding.getSessionContext());
+    }
+
+    private WriteConcern getAppliedWriteConcern(final AsyncWriteBinding binding) {
+        return getAppliedWriteConcern(binding.getSessionContext());
+    }
+
+    private WriteConcern getAppliedWriteConcern(final SessionContext sessionContext) {
+        if (sessionContext.hasActiveTransaction()) {
+            return WriteConcern.ACKNOWLEDGED;
+        } else {
+            return writeConcern;
+        }
+    }
+
+    private boolean shouldAcknowledge(final BulkWriteBatch batch, final SessionContext sessionContext) {
```

Comment:
After a few aborted attempts, this seemed the simplest way of fixing the bug: just make sure that everywhere we used to just use the write concern, use the default write concern when in a transaction.  This mimics the check in the BulkWriteBatch constructor, so it feels like the right level to do this at.